### ISSUE TITLE
Implement Basic Task Tracking Functionality

### ISSUE THREAD

**Issue by ejfox** [room302studio/coachartie#46]
Develop the capability for the AI assistant to record and store tasks. Tasks should be minimal with attributes such as title, description, status, and projectId to associate with specific projects

**Comment by rajivsinclair** [comment]
So that our todo items can be queried precisely and reliably, I want to be sure that we store each item as records in the database. 

In parallel with this project, we're also developing a simple front-end app that displays our todo list in a web UI which is integrated into Missive as a custom sidebar. This front-end will use the Supabase JS client to interact with the todos tables in our database. If you're curious about our sidebar project, [here](https://docs.pdw.co/QnoVTmV4cO0did) are some more details.

Here is a list of the attributes that we will want the ability to query on:
	- ID
	- Todo name
	- Description
	- Status (already done, in progress, todo)
	- Priority (Now, next, later)
	- Project ID
	- Project’s Missive conversation id 
	- Partner organization id
	- Related people/contacts array of ids
	- External URLs (e.g., linear or github issue or pull request)
	- Attachments (URLs of files in Supabase storage)
	- Who created this todo, or if Tachio created this todo on their own, then what is the reason that this todo was created
	- Which missive conversations are associated with this todo, array of conversation ids
	- Date-time when this todo was created
	- Date-time we started work on this todo
	- Date-time we completed work on this todo
	- Who is this todo assigned to (default is null, which means the whole team shares responsibility)
	- Parent, child, peer relationships to any other todo(s) that are related to this one
	- Vector embeddings of the todo name+description

**Comment by rajivsinclair** [comment]
I've created a branch with my first draft of the todo-related [capabilities](https://github.com/room302studio/coachartie/blob/2cc988403b0d35c8f7f9d7135962f7d99c6f3da9/capabilities/todos) (using the Supabase JS client to read/write on the various todos tables) and migrations for the [db](https://github.com/room302studio/coachartie/blob/2cc988403b0d35c8f7f9d7135962f7d99c6f3da9/db-migrations).

**Comment by rajivsinclair** [comment]
Hey, I just want to make sure that you all saw my comment yesterday with the definition of the attributes we want, including _embeddings_ of each todo.
